File: impl/server/src/routers/health.rs
use crate::response::{ApiResult, ok};
use axum::Json;
use serde::Serialize;

#[derive(Serialize)]
pub struct HealthData {
    pub status: &'static str,
}

pub async fn health() -> Json<ApiResult<HealthData>> {
    ok(HealthData { status: "ok" })
}

File: impl/server/src/routers/health.ts
// src/routers/health.ts
// 健康检查：GET /api/health → ok({ status: "ok" })。无鉴权（探活端点）。
import { Hono } from "hono";
import { attachErrorHandler, ok } from "../response";

export function healthRouter(): Hono {
  const app = new Hono();
  attachErrorHandler(app);

  app.get("/api/health", (c) => c.json(ok({ status: "ok" })));

  return app;
}
